File: Node/NodeTank/rosebot.js
const rosebotDriveSystem = require("./rosebotDriveSystem");
const rosebotServos = require("./rosebotServos");

class RoseBot {
    constructor() {
        // Drive motors
        this.driveSystem = new rosebotDriveSystem.DriveSystem();

        // Servos
        this.initializeServos();  // Needs to happen asynchronously so moved into a helper method.
    }

    async initializeServos() {
        const pca9685Driver = await rosebotServos.createPca9685Driver();  // Takes like 40 milliseconds
        this.armServos = new rosebotServos.ArmServos(pca9685Driver);
        this.gripperServo = new rosebotServos.GripperServo(pca9685Driver);
        this.cameraServo = new rosebotServos.CameraServo(pca9685Driver);
    }
}

module.exports = {
    RoseBot
}

File: InitialLearning/PythonLearning/FirestoreViaPython/tank/rosebot.py
import rosebot_drive_system
import rosebot_servos
import rosebot_sensors

class RoseBot:
  def __init__(self):
    self.drive_system = rosebot_drive_system.DriveSystem()
    self.servos = rosebot_servos.Servos()
    self.ultrasonic_sensor = rosebot_sensors.UltrasonicSensor()
    self.line_sensors = rosebot_sensors.LineSensors()
